packages/charts/chart-web-components/src/horizontal-bar-chart/horizontal-bar-chart.ts

```diff
@@ -0,0 +1,447 @@
+import { attr, FASTElement, observable } from '@microsoft/fast-element';
+import { create as d3Create, select as d3Select } from 'd3-selection';
+import { createTabster, getGroupper, getMover, getTabsterAttribute, TABSTER_ATTRIBUTE_NAME } from 'tabster';
+import { jsonConverter, SVG_NAMESPACE_URI, validateChartPropsArray } from '../utils/chart-helpers.js';
+import { ChartDataPoint, ChartProps, Variant } from './horizontal-bar-chart.options.js';
+
+// During the page startup.
+const tabsterCore = createTabster(window);
+getMover(tabsterCore);
+getGroupper(tabsterCore);
+
+/**
+ * A Horizontal Bar Chart HTML Element.
+ *
+ * @public
+ */
+export class HorizontalBarChart extends FASTElement {
+  /**
+   * The type of the element, which is always "horizontalbarchart".
+   * @see The {@link https://developer.mozilla.org/en-US/docs/Web/API/HTMLTextAreaElement/type | `type`} property
+   *
+   * @public
+   */
+  public get type(): 'horizontalbarchart' {
+    return 'horizontalbarchart';
+  }
+
+  /**
+   * @public
+   * The type of the dialog modal
+   */
+  @attr
+  public variant: Variant = Variant.AbsoluteScale;
```

Comment:
Is this something that can be defaulted _without_ being set? IE, JS assumes this as the default state as does styling? This will prevent reflecting attributes by default.

Reply:
fixed in #33264